app/src/main/java/org/jellyfin/androidtv/ui/startup/StartupActivity.kt

```diff
@@ -65,15 +66,29 @@ class StartupActivity : FragmentActivity() {
 	}
 
 	private fun start() {
-		if (application!!.currentUser != null && mediaManager.isPlayingAudio()) {
-			openNextActivity()
-		} else {
-			// Clear audio queue in case left over from last run
-			mediaManager.clearAudioQueue()
-			mediaManager.clearVideoQueue()
-			showServerList()
+		var isLoaded = false
+		sessionRepository.currentSession.observe(this) { session ->
+			if (session != null) {
+				Timber.i("Found a session in the session repository, waiting for the currentUser in the application class.")
+
+				supportFragmentManager.beginTransaction()
+					.replace(R.id.content_view, SplashFragment())
+					.commit()
+
+				application?.currentUserLiveData?.observe(this) { currentUser ->
+					Timber.i("CurrentUser changed to ${currentUser?.id} while waiting for startup.")
+
+					if (currentUser != null) openNextActivity()
```

Comment:
I've noticed an issue with this part. Sometimes the `currentSession` is set but the app will never set the `currentUserLiveData` value (or it fails and sets it to null).

I'm not sure yet how to fix this but if I think of something I'll probably put it in a separate PR.